pytorch_lightning/profiler/pytorch.py

```diff
@@ -528,3 +429,354 @@ def teardown(self, stage: Optional[str] = None) -> None:
+            self._schedule is not None and self._schedule._schedule == self._default_schedule()
+        )
+
+    @staticmethod
+    @lru_cache(1)
+    def _default_schedule() -> Optional[callable]:
+        if _KINETO_AVAILABLE:
+            # Those schedule defaults allow the profiling overhead to be negligible over training time.
+            return torch.profiler.schedule(wait=1, warmup=1, active=3)
+
+    def _get_profiler_class(self) -> Type:
+        return torch.profiler.profile if _KINETO_AVAILABLE else torch.autograd.profiler.profile
+
+    def _init_profiler(self, action_name: str) -> None:
+        if self._schedule is not None:
+            self._schedule.setup(action_name)
+
+        self._create_profilers()
+
+        profiler = self.profiler.__enter__()
+        if profiler is not None:
+            self.profiler = profiler
+
+        if self._parent_profiler is not None:
+            self._parent_profiler.__enter__()
+
+    def _stop_action(self, action_name: str) -> None:
+        # the default schedule requires a minimum of 5 steps to properly work: `wait=1, warmup=1, active=3`.
+        # otherwise, this will raise a `segmentation fault`.
+        if self._should_override_schedule():
+            warning_cache.warn(
+                "The PyTorch Profiler default schedule will be overridden as there is not enough "
+                "steps to properly record traces."
+            )
+            self._schedule = None
+            self.profiler.schedule = torch.profiler.profiler._default_schedule_fn
+
+        if self._schedule is not None:
+            self._schedule.pre_step(action_name)
+
+        def on_trace_ready(profiler):
+            if self.dirpath is not None:
+                if self._export_to_chrome:
+                    handler = tensorboard_trace_handler(
+                        self.dirpath, self._prepare_filename(action_name=action_name, extension="")
+                    )
+                    handler(profiler)
+
+                if self._export_to_flame_graph:
+                    path = os.path.join(
+                        self.dirpath, self._prepare_filename(action_name=action_name, extension=".stack")
+                    )
+                    profiler.export_stacks(path, metric=self._metric)
+            else:
+                rank_zero_warn("The PyTorchProfiler failed to export trace as `dirpath` is None")
+
+        if not self._has_on_trace_ready:
+            self.profiler.on_trace_ready = on_trace_ready
+
+        if self._schedule is not None:
+            self.profiler.step_num = self._schedule.num_step
+        self.profiler.step()
+
+    def _delete_profiler(self) -> None:
+        if self.profiler is not None:
+            self.profiler.__exit__(None, None, None)
+            self._cache_functions_events()
+            self.profiler = None
+
+        if self._schedule is not None:
+            self._schedule.reset()
+
+
+class PyTorchProfilerKineto(BasePyTorchProfiler):
```

Comment:
If this becomes the new PyTorchProfiler, we should just call it that IMO. Easier to remember

Reply:
are there any better way to make the PyTorchProfiler point to the legacy profiler if torch version is less than the specific version? 

It seems that flake8 complains "redefinition of unused 'PyTorchProfiler'" when I try to "`PyTorchProfiler = PyTorchProfilerLegacy`"